src/lang.rs

```diff
@@ -424,9 +426,12 @@ impl std::ops::Add for Obj {
     type Output = EvalResult;
 
     fn add(self, rhs: Self) -> Self::Output {
-        match (self.as_double()?, rhs.as_double()?) {
+        if !(&self, &rhs).coercible_to_numeric() {
+            return Err(Signal::Error(Error::CannotBeCoercedToNumerics));
+        }
+        match (self, rhs) {
             (Obj::Vector(l), Obj::Vector(r)) => Ok(Obj::Vector(l + r)),
-            _ => internal_err!(),
+            _ => unreachable!("coercible_to_numeric call before already checks this"),
```

Comment:
In general, try to use `internal_err!()` when things can return `EvalResult` to avoid the `panic!`. The macro can also accept a string to provide a more meaningful error message and returns a standard R error instead of crashing the runtime.


```suggestion
            _ => internal_err!("coercible_to_numeric call before already checks this"),
```